mcu-interface/examples/log_messages.rs

```diff
@@ -29,8 +30,12 @@ async fn main() -> Result<()> {
     };
 
     tokio::select! {
-        () = recv_fut => Ok(()),
-        result = tokio::signal::ctrl_c() => { println!("ctrl-c detected"); result.wrap_err("failed to listen for ctrl-c")}
-
+        () = recv_fut => (),
+        result = tokio::signal::ctrl_c() => {
+            println!("ctrl-c detected");
+            result.wrap_err("failed to listen for ctrl-c")?;
+        }
     }
+
+    task_handle.await.wrap_err("can task terminated uncleanly")
```

Comment:
`shouldn't we send to the one_shot channel kill in case recv_fut returns?`
You are right. Dropping the interface will do this. I've updated the example to fix it 👍